src/Microsoft.ML.KMeansClustering/KMeansPlusPlusTrainer.cs

```diff
@@ -45,34 +45,54 @@ public enum InitAlgorithm
         }
 
         [BestFriend]
-        internal static class Defaults{
+        internal static class Defaults
+        {
             /// <value>The number of clusters.</value>
-            public const int K = 5;
+            public const int ClustersCount = 5;
         }
 
-        public class Arguments : UnsupervisedLearnerInputBaseWithWeight
+        public class Options : UnsupervisedLearnerInputBaseWithWeight
         {
-            [Argument(ArgumentType.AtMostOnce, HelpText = "The number of clusters", SortOrder = 50)]
+            /// <summary>
+            /// The number of clusters.
+            /// </summary>
+            [Argument(ArgumentType.AtMostOnce, HelpText = "The number of clusters", SortOrder = 50, Name = "K")]
             [TGUI(SuggestedSweeps = "5,10,20,40")]
             [TlcModule.SweepableDiscreteParam("K", new object[] { 5, 10, 20, 40 })]
-            public int K = Defaults.K;
+            public int ClustersCount = Defaults.ClustersCount;
 
+            /// <summary>
+            /// Cluster initialization algorithm.
+            /// </summary>
             [Argument(ArgumentType.AtMostOnce, HelpText = "Cluster initialization algorithm", ShortName = "init")]
             public InitAlgorithm InitAlgorithm = InitAlgorithm.KMeansParallel;
 
+            /// <summary>
+            /// Tolerance parameter for trainer convergence. Low = slower, more accurate.
+            /// </summary>
             [Argument(ArgumentType.AtMostOnce, HelpText = "Tolerance parameter for trainer convergence. Low = slower, more accurate",
-                ShortName = "ot")]
+                Name = "OptTol", ShortName = "ot")]
```

Comment:
>"ot" [](start = 45, length = 4)

you can enumerate ShortNames  via comma "opttol,ot"

Reply:
Apparently name is for maml, and i set it to the previous name for backwards compatibility.

---
In reply to: [248405346](https://github.com/dotnet/machinelearning/pull/2158#discussion_r248405346) [](ancestors = 248405346)